packages/pubsub/src/subscription.js

```diff
@@ -592,30 +350,154 @@ Subscription.prototype.decorateMessage_ = function(message) {
  *   var apiResponse = data[0];
  * });
  */
-Subscription.prototype.delete = function(callback) {
+Subscription.prototype.delete = function(gaxOpts, callback) {
   var self = this;
 
-  callback = callback || common.util.noop;
+  if (is.fn(gaxOpts)) {
+    callback = gaxOpts;
+    gaxOpts = {};
+  }
 
-  var protoOpts = {
-    service: 'Subscriber',
-    method: 'deleteSubscription'
+  var reqOpts = {
+    subscription: this.name
   };
 
+  this.request({
+    client: 'subscriberClient',
+    method: 'deleteSubscription',
+    reqOpts: reqOpts,
+    gaxOpts: gaxOpts
+  }, function(err, resp) {
+    if (!err) {
+      self.closeConnection_();
```

Comment:
> For example, we could add 1 message listener, which would call self.openConnection_() as a consequence. We could then add 999 message listeners, none of which would call self.openConnection_().

I'm a little lost on why we would want to call `openConnection` when it's already open.

> Then we could call deleteSubscription, which would call self.closeConnection_(). Then we could add another message listener, which would call self.openConnection_() again. 

But this should throw an error if the subscription is deleted, right?

> We would then have 1000 message listeners hanging on to a closed connection.

We can easily kill off the event listeners, but I guess I'm confused why we think a user would delete a subscription and then hang on to it.

> So we called self.openConnection_() on the 1st and 1001st message listeners. That doesn't make too much sense.

I'm confused by this. Yes, we open on the 1st, because the user is looking for messages and a connection does not exist. Why would we open additional connections for all other event listeners?